### PR TITLE
METRON-2310: Remove metron-integration-test from compile dependencies

### PR DESCRIPTION
**Pull request by justinleet**
## Contributor Comments
Came out of debugging https://github.com/apache/metron/pull/1554, and was worth pulling out separately.

I had to add in `elasticsearch-rest-high-level-client` to `metron-elasticsearch-storm` since it seems that it was coming from `metron-integration-test` and indexing and REST both failed to deal with ES properly otherwise.

This seems to reduce the size of `metron-elasticsearch-storm` jar by ~20M.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

- [x] Have you ensured that any documentation diagrams have been updated, along with their source files, using [draw.io](https://www.draw.io/)? See [Metron Development Guidelines](https://cwiki.apache.org/confluence/display/METRON/Development+Guidelines) for instructions.

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

**Comment by justinleet** [comment]
Example of jar contents before and after

The resulting jar from this PR, which produces no results:
```
jar -tf metron-elasticsearch-storm-0.7.2-uber-shaded.jar| grep 'mockito'
```

From master:
```
jar -tf metron-elasticsearch-storm-0.7.2-uber-master.jar | grep 'mockito' | head -n 10
```

```
org/mockito/
org/mockito/AdditionalAnswers.class
org/mockito/AdditionalAnswers.java
org/mockito/AdditionalMatchers.class
org/mockito/AdditionalMatchers.java
org/mockito/Answers.class
org/mockito/Answers.java
org/mockito/ArgumentCaptor.class
org/mockito/ArgumentCaptor.java
org/mockito/ArgumentMatcher.class
```

**Comment by justinleet** [comment]
Looks like this occurred in https://github.com/apache/metron/pull/1389/

The dropping of `<scope>test</scope>` can be seen before:
https://github.com/apache/metron/pull/1389/files?file-filters%5B%5D=.xml#diff-15fdf86fc941b507912e1da249351370L155
and after:
https://github.com/apache/metron/pull/1389/files?file-filters%5B%5D=.xml#diff-1900ae3c8224fdd8f9c3b2bffce825edR49

File: metron-platform/metron-elasticsearch/metron-elasticsearch-storm/pom.xml [hunk 1]
```diff
@@ -36,6 +36,11 @@
             <artifactId>elasticsearch-shaded</artifactId>
             <version>${project.parent.version}</version>
         </dependency>
+        <dependency>
```
Comment (mmiklavc) [inline metron-platform/metron-elasticsearch/metron-elasticsearch-storm/pom.xml]: What's going on here - were we sneaking in the ES client dep via a transitive dep from the integration tests?

Comment (mmiklavc) [reply metron-platform/metron-elasticsearch/metron-elasticsearch-storm/pom.xml]: Nevermind - I dove into the code before reading your comments ;)

**Comment by mmiklavc** [comment]
Looks reasonable to me @justinleet, thanks for the fix! +1 by inspection.

**Pull request closed without merging**